common/src/main/java/org/conscrypt/NativeCrypto.java

```diff
@@ -535,7 +535,7 @@ static native long X509_get_notAfter(long x509ctx, OpenSSLX509Certificate holder
     static native byte[] X509_get_serialNumber(long x509ctx, OpenSSLX509Certificate holder);
 
     static native void X509_verify(long x509ctx, OpenSSLX509Certificate holder, NativeRef.EVP_PKEY pkeyCtx)
-            throws BadPaddingException;
+            throws BadPaddingException, IllegalBlockSizeException;
```

Comment:
nit: would it be easy to cover it with a test?

Reply:
Ought to be a straightforward variation on [this test](https://cs.android.com/android/platform/superproject/main/+/main:libcore/luni/src/test/java/libcore/java/security/cert/X509CertificateTest.java;l=347-356;drc=63b3644810104f37075345d29493962e1f07cc9f) where the second key is guaranteed to be too small. But I'm under a bit of time pressure to fix a flakiness issue so I'll raise a bug for it.